src/sha256.cpp

```diff
@@ -178,25 +193,25 @@ std::string sha256_hash::digest()
     num = 0;
     memset(p, 0, block_size);
 
-    std::array<char, 64> final_digest{0};
-    for (unsigned int nn = 0; nn < sha_digest_length; nn += 8) {
+    for (unsigned int nn = 0; nn < N; nn += 8) {
```

Comment:
A more readable variable name probably wouldn't hurt 😅

Reply:
I'll rename `N` to `DigestLength`, I want to keep the rest similar to OpenSSL.